q2cli/_tools.py

```diff
@@ -0,0 +1,89 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import zipfile
+import tempfile
+
+import click
+import qiime
+
+
+@click.group(help='Tools for working with QIIME files.')
+def tools():
+    pass
+
+
+@tools.command(short_help='View a QIIME Visualization.',
+               help="Displays a QIIME Visualization until the command exits. "
+                    "To open a QIIME Visualization so it can be used after "
+                    "the command exits, use 'qiime extract'.")
+@click.argument('visualization-path')
+@click.option('--index-extension', required=False, default='html',
+              help='The extension of the index file that should be opened. '
+                   '[default: html]')
+def view(visualization_path, index_extension):
+    if index_extension.startswith('.'):
+        index_extension = index_extension[1:]
+    try:
+        type_ = qiime.sdk.Result.peek(visualization_path).type
+    except zipfile.BadZipFile:
+        raise click.BadParameter(
+            '%s is not a QIIME Result. Only QIIME Visualizations can be '
+            'viewed.' % visualization_path)
+    # TODO: is there a better way to do this type check (without accessing
+    # qiime.core?)
+    if type_ != qiime.core.type.Visualization:
```

Comment:
Not at the moment, we might think about exposing types at something like `qiime.sdk.types` (in fact we would probably move `semantic.py`, `primitive.py` and `visualizations.py`, leaving `grammar.py` in core), but I think we need a larger conversation about package structure post-alpha.